File: 125-valid-palindrome/125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        left = 0
        right = len(s)-1
        
        while left < right:
            # find left char
            while not s[left].isalnum():
                if left == right:
                    break
                left += 1
            # find right char
            while not s[right].isalnum():
                if left == right:
                    break
                right -= 1
            
            if s[left].lower() != s[right].lower():
                return False
            
            left += 1
            right -= 1
        
        return True

File: turing_planet/string/125.valid-palindrome.cpp
/*
 * @lc app=leetcode id=125 lang=cpp
 *
 * [125] Valid Palindrome
 */

// @lc code=start
class Solution
{
  public:
    bool isPalindrome(string s)
    {
        int l = 0;
        int r = s.size() - 1;
        while (l < r)
        {
            // find first alphanum for l
            while (!isalnum(s[l]) && l < r)
                l++;
            while (!isalnum(s[r]) && l < r)
                r--;
            if (tolower(s[l]) != tolower(s[r]))
                return false;
            l++;
            r--;
        }
        return true;
    }
};
// @lc code=end
